### ISSUE TITLE
Default configs are present even when custom ones are defined

### ISSUE THREAD

**Issue by edobezabb** [ranyitz/vscode-search-presets#5]
As title says, the default configurations are showing up even if custom ones are defined.
I think this is unwanted behavior.

**Comment by ranyitz** [comment]
Hi @edobezabb! Can you please explain what do you mean? Ideally a list of steps, wanted behavior and actual behavior would help.

**Comment by GlebMorgan** [comment]
Hey @ranyitz, I think I can clarify 🙂.

There is no way to remove default presets (`Tests`, `Source Files` and `package.json`) from the list when searching presets with `Shift+Alt+F`. They are always present in the list of presets. It is only possible to add new presets via `search-presets.custom` configuration option. Moreover, if I want to have a custom preset named `Tests`, it is not possible to overwrite the default one.

I think it would be better if the extension did not have any default presets defined, or at least custom presets should overwrite default ones if they have the same name.

### Wanted behavior
1. It is possible to remove default preset (or all default presets) from the list of presets shown at `Shift+Alt+F`.
2. User-defined presets in `search-presets.custom` overwrite default presets supplied with the extension.

### Actual behavior
1. Default presets are always present and there is no way to remove them from the list of presets.
2. User-defined presets in `search-presets.custom` which have the same name as default presets do not appear in the list of presets.

**Comment by ranyitz** [comment]
Thanks for the explanation @GlebMorgan

**I now understand what's happening, let me explain**
1. We agree on the desired behavior, which is currently working correctly when installing the app.
2. I believe the issue involves the interaction between ["workspace settings" and "user settings"](https://code.visualstudio.com/docs/configure/settings). The product works such that workspace settings override user settings. So if you have defaults written to your workspace settings, any changes to the user settings for this particular project won't take effect.

**Why did this happen though? You probably didn't manually write anything to workspace settings yourself.**

It happened because of a feature that checks if this workspace has search-presets configuration and prompts you with a question asking if you want to set it up. After clicking OK, configuration is added to your workspace settings, which then overrides any configuration in user settings.

<img width="617" alt="Image" src="https://github.com/user-attachments/assets/db23d8ef-0ff5-465a-8547-a4da7baed2d1" />

**What I think we should do**
Configuring presets at the workspace level is a valuable feature if you know what you're doing, but I'm not certain it's how most people will use the extension. It seems the typical usage is to configure good cross-project defaults.

When I initially created the prompt to automatically set up workspace settings, I thought it would help users establish initial configuration, as writing it manually could be challenging. Now, considering this issue, I think I should approach this differently.

**Action items:**
1. When prompting the user, suggest saving the configuration to User settings if it doesn't exist there. I believe this was the original goal of the feature, and writing to workspace settings is actually a bug. This should be relatively straightforward to fix by [changing this line](https://github.com/ranyitz/vscode-search-presets/blob/d9e929410305126f1d20dc096b4a532e02682fe8/src/configuration/presets.ts#L45).
2. Consider implementing a command to copy existing search configuration as a preset (e.g., "Save as preset").

@GlebMorgan to verify, please check your workspace settings and see if removing the settings from workspace settings allows the user settings to take effect.

**Comment by GlebMorgan** [comment]
Hey, @ranyitz, thanks for a quick update!

You are right, I didn't notice that the extension modified my workspace settings (had a mess there because of other issues), my bad.
After removing default presets from workspace settings only ones configured in user settings remained.

The reason I'm modifying user settings and not workspace is simple – I'm working on a shared repository and workspace settings are under git.

This is resolved for me, hope this also helps @edobezabb 🙂

P.S. I also think that adding default presets to user settings is a better choice compared to workspace, but this is not that important.

**Comment by ranyitz** [comment]
> I also think that adding default presets to user settings is a better choice compared to workspace, but this is not that important.

You're right, that was a bug, which is now resolved in the latest version of the extension.
Thank you for raising the issue 🙏

**Comment by ranyitz** [comment]
@GlebMorgan I also added a nicer way to add configuration via UI, it will add it to the user settings - https://github.com/ranyitz/vscode-search-presets/pull/7
